### PR TITLE
Adding Automation to MAISTRA-2150

### PR DESCRIPTION
**Pull request by gbaufake**
Test Run on the fixed version:

```
 go test -run 31 -timeout 1m -v
Preparing environment configurations. Wait to start...
=== RUN   31
2021-04-09T17:59:42.423043Z     info    Running command oc rsh -c discovery $(oc get pods -l app=pilot --no-headers | awk '{print $1}') curl -v http://localhost:8080/debug/cdsz
2021-04-09T17:59:46.214553Z     info    Command output: 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /debug/cdsz HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.61.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Fri, 09 Apr 2021 17:59:46 GMT
< Transfer-Encoding: chunked
< 
{ [3052 byte[


  {"node": "sidecar~10.129.2.70~reviews-v1-99f674f45-bgp92.bookinfo-v11~bookinfo-v11.svc.cluster.local-57", "addr": "10.129.2.70:53096", "connect": "2021-04-09 17:56:34.847246847 +0000 UTC m=+5579.581626451","Clusters":[
]}
,


  {"node": "sidecar~10.131.0.54~reviews-v3-5bb8d84b4f-mqf4v.bookinfo-v11~bookinfo-v11.svc.cluster.local-50", "addr": "10.131.0.54:59744", "connect": "2021-04-09 17:30:54.49420046 +0000 UTC m=+4039.228580216","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.69~details-v1-5447bb8796-fmp8x.bookinfo-v11~bookinfo-v11.svc.cluster.local-60", "addr": "10.129.2.69:48456", "connect": "2021-04-09 17:59:43.177761787 +0000 UTC m=+5767.912141402","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.32~mysqldb-v1-86484d5dbd-g95k4.bookinfo-v11~bookinfo-v11.svc.cluster.local-56", "addr": "10.128.2.32:44056", "connect": "2021-04-09 17:55:36.975223105 +0000 UTC m=+5521.709602952","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.35~reviews-v2-7f866f9d65-pklmf.bookinfo-v11~bookinfo-v11.svc.cluster.local-59", "addr": "10.128.2.35:37724", "connect": "2021-04-09 17:57:17.825261168 +0000 UTC m=+5622.559640924","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.71~kiali-traffic-generator-6756b666f5-pzpfk.bookinfo-v11~bookinfo-v11.svc.cluster.local-54", "addr": "10.129.2.71:47738", "connect": "2021-04-09 17:34:17.156656503 +0000 UTC m=+4241.891036296","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.67~mongodb-v1-6cdd79449d-m6ffb.bookinfo-v11~bookinfo-v11.svc.cluster.local-47", "addr": "10.129.2.67:46098", "connect": "2021-04-09 17:28:39.968629955 +0000 UTC m=+3904.703009575","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.68~ratings-v2-547f686889-gn9rr.bookinfo-v11~bookinfo-v11.svc.cluster.local-48", "addr": "10.129.2.68:46760", "connect": "2021-04-09 17:30:25.302086133 +0000 UTC m=+4010.036465946","Clusters":[
]}
,


  {"node": "sidecar~10.131.0.52~ratings-v2-mysql-7c4654c8dd-8phzg.bookinfo-v11~bookinfo-v11.svc.cluster.local-51", "addr": "10.131.0.52:41536", "connect": "2021-04-09 17:31:21.001733696 +0000 UTC m=+4065.736113299","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.36~productpage-v1-6c9f9f7b98-gjrqk.bookinfo-v11~bookinfo-v11.svc.cluster.local-53", "addr": "10.128.2.36:35146", "connect": "2021-04-09 17:32:22.651414923 +0000 UTC m=+4127.385794677","Clusters":[
]}
,


  {"node": "router~10.129.2.62~istio-ingressgateway-6db6d57c74-b5bwj.istio-system-v11~istio-system-v11.svc.cluster.local-55", "addr": "10.129.2.62:36268", "connect": "2021-04-09 17:48:01.656584484 +0000 UTC m=+5066.390964191","Clusters":[
]}
,


  {"node": "sidecar~10.131.0.53~ratings-v1-7dd8bcffdd-mj76d.bookinfo-v11~bookinfo-v11.svc.cluster.local-52", "addr": "10.131.0.53:51582", "connect": "2021-04-09 17:31:45.608191959 +0000 UTC m=+4090.342571764","Clusters":[
]}
,


  {"node": "router~10.129.2.60~istio-egressgateway-5c6ff46c98-q76fl.istio-system-v11~istio-system-v11.svc.cluster.local-58", "addr": "10.129.2.60:36034", "connect": "2021-04-09 17:56:49.861313522 +0000 UTC m=+5594.595693227","Clusters":[
]}
]
s data]
100  3033    0  3033    0     0  2961k      0 --:--:-- --:--:-- --:--:-- 2961k
* Connection #0 to host localhost left intact
2021-04-09T17:59:46.214772Z     info    Running command oc rsh -c discovery $(oc get pods -l app=pilot --no-headers | awk '{print $1}') curl -v http://localhost:8080/debug/cdsz
2021-04-09T17:59:46.214954Z     info    Running command PILOT=$(oc get pods -l app=pilot --no-headers | awk '{print $1}') ; count=1 ; while : ; do oc rsh -c discovery ${PILOT} curl -o /dev/null -s -w "%!{(MISSING)http_code}
" http://localhost:8080/debug/edsz ; echo $(date): loop $count ; count=$(($count + 1)); done
2021-04-09T17:59:46.214988Z     info    Running command count=1 ; while : ; do oc rollout restart deployment istio-ingressgateway >/dev/null ; echo $(date): Restarted count $count ; count=$(($count + 1)); sleep 5 ; done
2021-04-09T17:59:49.749075Z     info    Command output: 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /debug/cdsz HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.61.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Fri, 09 Apr 2021 17:59:49 GMT
< Transfer-Encoding: chunked
< 
{ [3052 bytes data]
100  3033    0  3033    0     0  2961k      0 --:--:-- --:--:-- --:--:-- 2961k
* Connection #0 to host localhost left intact
[


  {"node": "sidecar~10.131.0.53~ratings-v1-7dd8bcffdd-mj76d.bookinfo-v11~bookinfo-v11.svc.cluster.local-52", "addr": "10.131.0.53:51582", "connect": "2021-04-09 17:31:45.608191959 +0000 UTC m=+4090.342571764","Clusters":[
]}
,


  {"node": "router~10.129.2.60~istio-egressgateway-5c6ff46c98-q76fl.istio-system-v11~istio-system-v11.svc.cluster.local-58", "addr": "10.129.2.60:36034", "connect": "2021-04-09 17:56:49.861313522 +0000 UTC m=+5594.595693227","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.70~reviews-v1-99f674f45-bgp92.bookinfo-v11~bookinfo-v11.svc.cluster.local-57", "addr": "10.129.2.70:53096", "connect": "2021-04-09 17:56:34.847246847 +0000 UTC m=+5579.581626451","Clusters":[
]}
,


  {"node": "sidecar~10.131.0.54~reviews-v3-5bb8d84b4f-mqf4v.bookinfo-v11~bookinfo-v11.svc.cluster.local-50", "addr": "10.131.0.54:59744", "connect": "2021-04-09 17:30:54.49420046 +0000 UTC m=+4039.228580216","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.69~details-v1-5447bb8796-fmp8x.bookinfo-v11~bookinfo-v11.svc.cluster.local-60", "addr": "10.129.2.69:48456", "connect": "2021-04-09 17:59:43.177761787 +0000 UTC m=+5767.912141402","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.32~mysqldb-v1-86484d5dbd-g95k4.bookinfo-v11~bookinfo-v11.svc.cluster.local-56", "addr": "10.128.2.32:44056", "connect": "2021-04-09 17:55:36.975223105 +0000 UTC m=+5521.709602952","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.35~reviews-v2-7f866f9d65-pklmf.bookinfo-v11~bookinfo-v11.svc.cluster.local-59", "addr": "10.128.2.35:37724", "connect": "2021-04-09 17:57:17.825261168 +0000 UTC m=+5622.559640924","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.71~kiali-traffic-generator-6756b666f5-pzpfk.bookinfo-v11~bookinfo-v11.svc.cluster.local-54", "addr": "10.129.2.71:47738", "connect": "2021-04-09 17:34:17.156656503 +0000 UTC m=+4241.891036296","Clusters":[
]}
,


  {"node": "sidecar~10.128.2.36~productpage-v1-6c9f9f7b98-gjrqk.bookinfo-v11~bookinfo-v11.svc.cluster.local-53", "addr": "10.128.2.36:35146", "connect": "2021-04-09 17:32:22.651414923 +0000 UTC m=+4127.385794677","Clusters":[
]}
,


  {"node": "router~10.129.2.62~istio-ingressgateway-6db6d57c74-b5bwj.istio-system-v11~istio-system-v11.svc.cluster.local-55", "addr": "10.129.2.62:36268", "connect": "2021-04-09 17:48:01.656584484 +0000 UTC m=+5066.390964191","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.67~mongodb-v1-6cdd79449d-m6ffb.bookinfo-v11~bookinfo-v11.svc.cluster.local-47", "addr": "10.129.2.67:46098", "connect": "2021-04-09 17:28:39.968629955 +0000 UTC m=+3904.703009575","Clusters":[
]}
,


  {"node": "sidecar~10.129.2.68~ratings-v2-547f686889-gn9rr.bookinfo-v11~bookinfo-v11.svc.cluster.local-48", "addr": "10.129.2.68:46760", "connect": "2021-04-09 17:30:25.302086133 +0000 UTC m=+4010.036465946","Clusters":[
]}
,


  {"node": "sidecar~10.131.0.52~ratings-v2-mysql-7c4654c8dd-8phzg.bookinfo-v11~bookinfo-v11.svc.cluster.local-51", "addr": "10.131.0.52:41536", "connect": "2021-04-09 17:31:21.001733696 +0000 UTC m=+4065.736113299","Clusters":[
]}
]
--- PASS: 31 (7.33s)
PASS
ok      github.com/Maistra/maistra-test-tool    58.612s

```

### CODE REVIEW

File: tests/smcp_recursive_lock_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+package tests
+
+import (
+	"maistra/util"
+	"testing"
+)
+
+func testPilotLock() {
+	if _ , err := util.Shell("oc rsh -c discovery $(oc get pods -l app=pilot --no-headers | awk '{print $1}') curl -v http://localhost:8080/debug/cdsz") ; err != nil {
```
Comment (yxun) [inline tests/smcp_recursive_lock_test.go:23]: @gbaufake are those oc rsh and oc get command supposed to be run in istio-system ns ?  Could you update the commands with -n istio-system ?

Comment (gbaufake) [reply tests/smcp_recursive_lock_test.go:23]: Sure!

File: tests/smcp_recursive_lock_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+}
+
+
+func queryPilot() {
+	util.Shell("PILOT=$(oc get pods -l app=pilot --no-headers | awk '{print $1}') ; count=1 ; while : ; do oc rsh -c discovery ${PILOT} curl -o /dev/null -s -w \"%{http_code}\n\" http://localhost:8080/debug/edsz ; echo $(date): loop $count ; count=$(($count + 1)); done")
+}
+
+func rolloutIngressGateway() {
+	util.Shell("count=1 ; while : ; do oc rollout restart deployment istio-ingressgateway >/dev/null ; echo $(date): Restarted count $count ; count=$(($count + 1)); sleep 5 ; done")
```
Comment (yxun) [inline tests/smcp_recursive_lock_test.go:34]: Same here. It's better to add -n istio-system when restarting istio-ingressgateway deployment.

**Comment by gbaufake** [comment]
@yxun 

Just pushed the changes with `istio-system` namespace

**Review by yxun — APPROVED** [review]

**Pull request merged**